annotations/views.py

```diff
@@ -364,6 +366,62 @@ def collection_texts(request, collectionid):
     return HttpResponse(template.render(context))
 
 
+class RecentActivity(models.Model):
+    def __init__(self):
+        self.appellation = 0
+        self.text = 0
+
+
+def recent_activity(request):
+    from datetime import datetime, timedelta
+    from django.utils import timezone
+    template = loader.get_template('annotations/recent_activity.html')
+    time_threshold = datetime.now() - timedelta(days=1)
+    text_list = list(Text.objects.filter(added__gt=time_threshold).order_by('-added')[:10])
+
+    activity_data={}
+
+    if text_list:
+        for initialHour in range(1, 25):
+            for item in text_list:
+                username = item.addedBy.username
+                if not activity_data.get(initialHour):
+                    activity_data[initialHour] = {}
+                if item.added > timezone.now() - timedelta(hours=initialHour):
+                    userDict = activity_data[initialHour]
+                    if not userDict.get(username):
+                        activity = RecentActivity()
+                        activity.text +=1
+                        userDict[username] = activity
+                    else:
+                        activity = userDict.get(username)
+                        activity.text +=1
+                text_list.remove(item)
+
+    appellation_list = list(Appellation.objects.filter(created__gt=time_threshold).order_by('-created')[:10])
```

Comment:
We took only 10 recent records to minimize the number of stories we need to show in recent activity. Since as per our initial discussion we needed to only consider 5 recent stories.
We can tweak the number as per our need.